test/spec/AuthStateManager.js

```diff
@@ -276,11 +286,19 @@ describe('AuthStateManager', () => {
       expect(handler).toHaveBeenCalledTimes(1);
     });
 
-    it('should only trigger authStateManager.updateAuthState once when call tokenManager.add', () => {
+    it('evaluates authState based on the latest token event', async (done) => {
       const auth = createAuth();
-      auth.authStateManager.updateAuthState = jest.fn();
+      jest.spyOn(auth.authStateManager, 'updateAuthState');
+      // add tokens async
       auth.tokenManager.add('idToken', tokens.standardIdTokenParsed);
-      expect(auth.authStateManager.updateAuthState).toHaveBeenCalledTimes(1);
+      await wait(100);
```

Comment:
what is special about 100ms? Can we use fake timers to better illustrate what we are trying to test?

Reply:
before we added p-cancelable to avoid very close promises, here the `wait(100)` makes the assumption that the two consective promise should be be cancelled, but be debounced instead.